DEVELOPERS.md

```diff
@@ -1,225 +1,205 @@
-If you do not have installed poetry yet, it is recommended to
-[install it with pipx](https://python-poetry.org/docs/#installing-with-pipx).
+You will need to [install conda](https://docs.anaconda.com/free/anaconda/install/) to use this method.
+We also recommend to install mamba in your base conda environment and use that command instead of conda.
+This will make the installation of the dependencies much faster.
 
-Additionally, it requires the following poetry plugins to be installed:
+```shell
+conda activate base
+conda install mamba -c conda-forge
+```
 
- - poetry-plugin-export
- - poetry-dynamic-versioning
+All base dependencies are included in the `environment.yml` file provided in the repo root.
+You can now run the following commands to create your pixelator environment:
 
-These can be installed using following commands:
+```shell
+conda env create -f environment.yml
+```
 
+Activate the conda environment:
 ```shell
-poetry self add poetry-plugin-export
-poetry self add poetry-dynamic-versioning[plugin]
+conda activate pixelator
 ```
 
-You also need to install the [fastp](https://github.com/OpenGene/fastp) program.
+### 3. Install pixelator
 
-- Clone the repository, navigate to the project root and checkout `dev`.
+To install pixelator you can now run:
 
-    ```shell
-    git@github.com:PixelgenTechnologies/pixelator.git
-    cd pixelator
-    git checkout dev
-    ```
+```shell
+task install
+```
 
-- We recommend that you use a poetry virtual environment in the root directory of the pixelator repository.
-You can do this be running the following command in the repository root:
+This will install pixelator with all it's dependencies in your virtual environment, and
+setup git hooks to run pre-commit checks before you commit your changes.
 
-    ```shell
-    poetry config --local virtualenvs.in-project true
-    ```
+You are now ready to start developing pixelator. Congrats!
 
-- Now install the project from the repository root using.
+## Pixelator developer tools
 
-    ```shell
-    poetry install
-    ```
+Pixelator provides a number of utility scripts (you have already used one above) to help with development and testing using [Task](https://taskfile.dev/).
 
-- If you want to activate a shell within your local env you can run:
-    ```shell
-    source .venv/bin/activate
-    ```
+> [!NOTE]
+> If you have followed the installation instructions above you should have `task` installed
+> in your virtual environment.
+> Otherwise you can follow the installation instructions here [here](https://taskfile.dev/#/installation).
 
-## Update poetry.lock
+> [!TIP]
+> Depending on the installation method the task executable might be named `task` or `go-task`.
 
-Currently, the `poetry.lock` file is in the source code to ensure the consistence
-with package versions. If you need to add or update a package in `pyproject.toml`
-you must then update and commit the changes in the `poetry.lock` file to the repository.
+### Viewing the taskfile
```

Comment:
Up until this point, the instructions are crystal clear. But I don't really know what task is used for. Would it be possible to add a sentence or two describing why we are using task?

Reply:
Good point. I will try to clarify this.